.circleci/config.yml

```diff
@@ -127,29 +132,37 @@ jobs:
           command: |
             VARS="-var tags={\"build_url\":\"$CIRCLE_BUILD_URL\"}"
             VARS+=' -var launch_type=<<parameters.launch_type>>'
+            VARS+=' -var consul_version=<<parameters.consul_version>>'
             case $CIRCLE_BRANCH in
-                main | release/*) VARS+=" -var enable_hcp=true";;
+                main | release/*) VARS+=" -var enable_hcp=<<parameters.enable_hcp>>";;
                 *) VARS+=" -var enable_hcp=false";;
             esac
 
             terraform destroy -auto-approve $VARS
           when: always
 
+
+acceptance-common: &acceptance-common
+  filters:
+    branches:
+      # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
+      ignore: /pull\/[0-9]+|docs\/.*/
+  requires:
+    - go-fmt-and-lint-acceptance
+
 workflows:
   version: 2.1
   test-and-build:
     jobs:
       - go-fmt-and-lint-acceptance
       - terraform-fmt
-      - acceptance:
-          matrix:
-            parameters:
-              launch_type:
-                - FARGATE
-                - EC2
-          filters:
-            branches:
-              # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
-              ignore: /pull\/[0-9]+|docs\/.*/
-          requires:
-            - go-fmt-and-lint-acceptance
+      # We have a limit of 6 HCP Consul clusters.
+      # The following controls whether to enable HCP when testing release branches.
+      # HCP is always disabled for tests on PRs.
+      - acceptance: {name: "acceptance-1.12-FARGATE-HCP", consul_version: '1.12.7', enable_hcp: true, launch_type: FARGATE, <<: *acceptance-common}
+      - acceptance: {name: "acceptance-1.13-FARGATE", consul_version: '1.13.4', enable_hcp: false, launch_type: FARGATE, <<: *acceptance-common}
+      - acceptance: {name: "acceptance-1.14-FARGATE-HCP", consul_version: '1.14.1', enable_hcp: true, launch_type: FARGATE, <<: *acceptance-common}
+      - acceptance: {name: "acceptance-1.12-EC2", consul_version: '1.12.7', enable_hcp: false, launch_type: EC2, <<: *acceptance-common}
+      - acceptance: {name: "acceptance-1.13-EC2-HCP", consul_version: '1.13.4', enable_hcp: true, launch_type: EC2, <<: *acceptance-common}
+      - acceptance: {name: "acceptance-1.14-EC2", consul_version: '1.14.1', enable_hcp: false, launch_type: EC2, <<: *acceptance-common}
```

Comment:
Here, I didn't use the full matrix of `launch_type x consul_version` because that would require all 6 of our HCP clusters, per release branch.

I settled for 3 HCP clusters at a time, so this will test all three Consul versions. Two of those HCP enabled test runs use Fargate, and one uses EC2. (Note that we always test with the dev server in each job, even when HCP is enabled)

For the config, the alternative is to use [matrix](https://circleci.com/docs/configuration-reference/#matrix) + excludes. But I felt that being explicit about what we _are_ testing is a little clearer (but uglier) versus figuring out all the configurations we need to exclude. Also, we don't need to run that same config twice with both HCP enabled and disabled (because we always run the dev server tests regardless).

Reply:
Not sure if it's coincidence, but the `acceptance-1.14-EC2` job has flaked several times. A couple times exhibited a CloudWatch parsing bug, which I've fixed - but this might indicate some underlying issue. Another time it failed seemingly without a specific error that I could find. Another time it showed an execute-command session timeout, but I was unsure of the cause.

But when I retried the 1.14 EC2 job, it passed each time. Also, the 1.14 Fargate job never failed. I've also run that test several times locally, and never saw a failure. 🤔 

I have an inkling there might be some kind of AWS limit we are hitting, either in session manager or in cloud watch. Maybe more parallel things happening (6 jobs * 3 parallel test cases) when all 6 jobs are running, compared to when I re-run a single job or locally in a test run.

Reply:
I saw the flakes (I even restarted one of the jobs). It does **seem** to just be flakes because it is not consistently failing for any specific reason.